### PR TITLE
Add spec for NOT opcode

### PR DESCRIPTION
**Pull request by z2trillion**

### CODE REVIEW

**Comment by icemelon** [comment]
@ed255 This PR should be ready for review now.

**Review by CPerezz — APPROVED** [review]
LGTM.

File: specs/opcode/19NOT.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+## Constraints
+
+1. opcodeId checks
+   - opId == OpcodeId(0x19)
+2. state transition:
+   - gc + 2
+   - stack_pointer unchanged
+   - pc + 1
+   - gas + 3
+3. Lookups: 34 busmapping lookups
```
Comment (ed255) [inline specs/opcode/19NOT.md:19]: Maybe better say: 2 busmapping lookups + 32 fixed table lookups?

File: src/zkevm_specs/evm/typing.py [hunk 2]
```diff
@@ -204,6 +204,9 @@ def __init__(self, code: Optional[bytearray] = None) -> None:
         self.code = bytearray() if code is None else code
 
     def __getattr__(self, name: str):
+        if name.startswith("bitwise_"):
```
Comment (ed255) [inline src/zkevm_specs/evm/typing.py:207]: notice that on line 212 there is:
```python
opcode = Opcode[name.removesuffix("_").upper()]
```
which allows you to do `Bytecode().not_(a)`

**Review by ed255 — COMMENTED** [review]
Looks good overall! Please take a look at my comments :)

File: src/zkevm_specs/evm/typing.py (hunk 2 above)
Comment (icemelon) [reply src/zkevm_specs/evm/typing.py:207]: good idea. will change it.

Comment (icemelon) [reply src/zkevm_specs/evm/typing.py]: fixed in 5b9a0d3

File: specs/opcode/19NOT.md (hunk 1 above)
Comment (icemelon) [reply specs/opcode/19NOT.md]: fixed in 5b9a0d3

**Review by ed255 — APPROVED** [review]
LGTM!

**Pull request merged**